Simplenote/TagListViewController.swift

```diff
@@ -376,89 +372,104 @@ private extension TagListViewController {
         cell.hasClearBackground = true
         cell.imageTintColor = .simplenoteTintColor
     }
+}
 
-    func shouldSelectCell(at indexPath: IndexPath) -> Bool {
-        guard let section = Section(rawValue: indexPath.section) else {
-            return false
+// MARK: - Selection
+//
+private extension TagListViewController {
+    func updateCurrentSelection() {
+        tableView.selectRow(at: currentSelection, animated: false, scrollPosition: .none)
+    }
+
+    var currentSelection: IndexPath? {
+        let filter = NotesListFilter(selectedTag: SPAppDelegate.shared().selectedTag)
+        switch filter {
+        case .everything:
+            return IndexPath(row: SystemRow.allNotes.rawValue,
+                             section: Section.system.rawValue)
+        case .deleted:
+            return IndexPath(row: SystemRow.trash.rawValue,
+                             section: Section.system.rawValue)
+        case .untagged:
+            return IndexPath(row: 0,
+                             section: Section.bottom.rawValue)
+        case .tag(let tagName):
+            guard let index = resultsController.fetchedObjects.firstIndex(where: {
+                $0.name == tagName
+            }) else {
+                return nil
+            }
+
+            return IndexPath(row: index,
+                             section: Section.tags.rawValue)
         }
```

Comment:
Nipticky: (please feel free to disregard, just brainstorming!!)

```
private extension NotesListFilter {
    var indexPath: IndexPath
}
```

Reply:
We can't do that, because we use `resultsController` to get correct index for a tag

Reply:
Ahhhh totally missed that, apologies about the noise!!